src/project/manifest/document.rs

```diff
@@ -142,100 +132,104 @@ impl ManifestSource {
         dep: &PyPiPackageName,
         platform: Option<Platform>,
         feature_name: &FeatureName,
-    ) -> Result<(), Report> {
-        match self {
-            ManifestSource::PixiToml(_) => self
-                .remove_dependency_helper(
-                    dep.as_source(),
-                    consts::PYPI_DEPENDENCIES,
-                    platform,
-                    feature_name,
-                )
-                .map(|_| ()),
-            ManifestSource::PyProjectToml(_) => {
```

Comment:
By only testing on strings, this was removing from the TOML array any dependency containing the same name